### ISSUE TITLE
Cover explicit return for `invisible()` in `function_return_linter()`

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#2560]
Although the documentation for `invisible()` mentions

> Return a (temporarily) invisible copy of an object

it will actually return only if it's the _last_ line in the function, otherwise the function is going to proceed with the rest of the code. 

``` r
f <- function(x = NULL) {
  if (is.null(x)) {
    invisible(NA) # should lint and suggest to use `return(invisible(NA))`
  }
  
  "a"
}

x <- f()
x
#> [1] "a"
```

<sup>Created on 2024-04-19 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

I think this might be a good special case to consider in `function_return_linter()`.

**Comment by MichaelChirico** [comment]
My guess would be it's more likely the author just doesn't understand how `invisible()` works, they might not have meant for it to `return()` at all, or possibly it's just copy-pasted from a refactor where it used to be the return object & now it's not.

That said, the logic to find the "terminal" vs. "non-terminal" call of a function will be relevant.

PS I guess you mean `return_linter()`, `function_return_linter()` is for usage like `return(a <- foo())`, though we might want to consider subsuming the latter into the former...

**Comment by IndrajeetPatil** [comment]
> though we might want to consider subsuming the latter into the former

Yeah, I think we discussed this at some point. It totally makes sense to combine them into a single linter!
